### PR TITLE
GLT-2558: bulk remove and discard box items

### PR DESCRIPTION
**Pull request by djcooke**

### CODE REVIEW

**Comment by sqwci** [comment]
SonarQube analysis reported 6 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [BoxRestController.java#L508](https://github.com/TGAC/miso-lims/blob/44da66a64d13b3f91ee17315e2e40bfcb7770a67/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/BoxRestController.java#L508): Refactor this method to reduce its Cognitive Complexity from 22 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
1. ![CRITICAL][CRITICAL] [BoxRestController.java#L655](https://github.com/TGAC/miso-lims/blob/44da66a64d13b3f91ee17315e2e40bfcb7770a67/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/BoxRestController.java#L655): Define a constant instead of duplicating this literal " not found" 3 times. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)
1. ![MAJOR][MAJOR] [BoxRestController.java#L566](https://github.com/TGAC/miso-lims/blob/44da66a64d13b3f91ee17315e2e40bfcb7770a67/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/BoxRestController.java#L566): Extract this nested try block into a separate method. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1141)
1. ![MINOR][MINOR] [BoxRestController.java#L377](https://github.com/TGAC/miso-lims/blob/44da66a64d13b3f91ee17315e2e40bfcb7770a67/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/BoxRestController.java#L377): Combine this catch with the one at line 375, which has the same body. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2147)
1. ![MINOR][MINOR] [editBox.jsp#L145](https://github.com/TGAC/miso-lims/blob/44da66a64d13b3f91ee17315e2e40bfcb7770a67/miso-web/src/main/webapp/pages/editBox.jsp#L145): Add an "alt" attribute to this image. [rule](https://sonarcloud.io/coding_rules#rule_key=Web%3AImgWithoutAltCheck)
1. ![MINOR][MINOR] [editBox.jsp#L239](https://github.com/TGAC/miso-lims/blob/44da66a64d13b3f91ee17315e2e40bfcb7770a67/miso-web/src/main/webapp/pages/editBox.jsp#L239): Add an "alt" attribute to this image. [rule](https://sonarcloud.io/coding_rules#rule_key=Web%3AImgWithoutAltCheck)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/BoxRestController.java [hunk 1]
```diff
@@ -630,6 +630,25 @@ private static boolean isRealBarcode(BoxScan scan, String barcode) {
     return getBoxDtoWithBoxables(boxId);
   }
 
+  @PostMapping("/box/{boxId}/bulk-remove")
+  public @ResponseBody BoxDto removeMultipleItems(@PathVariable long boxId, @RequestBody List<String> positions) throws IOException {
+    Box box = getBox(boxId);
+    for (String position : positions) {
+      box.getBoxPositions().remove(position);
```
Comment (callunity) [inline miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/BoxRestController.java]: Could you add a check that there is an item in the position before removing it? (I selected a column that had a few blank positions and clicked "discard selected" and it returned `Internal Server Error: No item in the specified box position` and when I closed the dialog the page didn't refresh so it looked like nothing had happened, but when I manually refreshed the page it turned out that the selected items had actually been discarded)

Comment (djcooke) [reply miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/BoxRestController.java]: Oops, yes. I was originally returning 300 bad request if a position was empty, but that was pointless/annoying. Forgot to still check. Fixed now

**Comment by sqwci** [comment]
SonarQube analysis reported 6 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [BoxRestController.java#L508](https://github.com/TGAC/miso-lims/blob/a64d59012c7ca3cbe5c222d8ef532e1ee25fabe6/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/BoxRestController.java#L508): Refactor this method to reduce its Cognitive Complexity from 22 to the 15 allowed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)
1. ![CRITICAL][CRITICAL] [BoxRestController.java#L659](https://github.com/TGAC/miso-lims/blob/a64d59012c7ca3cbe5c222d8ef532e1ee25fabe6/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/BoxRestController.java#L659): Define a constant instead of duplicating this literal " not found" 3 times. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)
1. ![MAJOR][MAJOR] [BoxRestController.java#L566](https://github.com/TGAC/miso-lims/blob/a64d59012c7ca3cbe5c222d8ef532e1ee25fabe6/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/BoxRestController.java#L566): Extract this nested try block into a separate method. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1141)
1. ![MINOR][MINOR] [BoxRestController.java#L377](https://github.com/TGAC/miso-lims/blob/a64d59012c7ca3cbe5c222d8ef532e1ee25fabe6/miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/BoxRestController.java#L377): Combine this catch with the one at line 375, which has the same body. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2147)
1. ![MINOR][MINOR] [editBox.jsp#L145](https://github.com/TGAC/miso-lims/blob/a64d59012c7ca3cbe5c222d8ef532e1ee25fabe6/miso-web/src/main/webapp/pages/editBox.jsp#L145): Add an "alt" attribute to this image. [rule](https://sonarcloud.io/coding_rules#rule_key=Web%3AImgWithoutAltCheck)
1. ![MINOR][MINOR] [editBox.jsp#L239](https://github.com/TGAC/miso-lims/blob/a64d59012c7ca3cbe5c222d8ef532e1ee25fabe6/miso-web/src/main/webapp/pages/editBox.jsp#L239): Add an "alt" attribute to this image. [rule](https://sonarcloud.io/coding_rules#rule_key=Web%3AImgWithoutAltCheck)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**